spirv_common.hpp

```diff
@@ -296,6 +296,8 @@ struct SPIRType : IVariant
 		bool depth;
 		bool arrayed;
 		bool ms;
+		bool is_read = false;
```

Comment:
`Image` `is_read` and `is_written` track whether the image is read or written to from within the shader. Under Metal, texture types need to be declared as writable in order to be written to. The `is_read` allows the converter to distinguish whether a writable texture should be declared with `write` or `read_write`...in case there is any efficiency to be gained by the Metal compiler by distinguishing between those two states.